torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -500,6 +503,10 @@ class TORCH_API Reducer {
 
   // comm_hook_ is used to access the DDP communication hook if registered.
   std::unique_ptr<CommHookInterface> comm_hook_;
+
+  // sparse metadata
+  std::unique_ptr<std::map<std::string, at::Tensor>> sparse_metadata_;
```

Comment:
nit: I would add a description of what the key here is and what the value is conceptually (i.e. I feel like `global_unique_id` is an unfamiliar term to most users). The API is private for now but might be good for conceptual undestanding.